### PR TITLE
Add content to readme

### LINKED ISSUE #728: Consider changing about and readme text
For users unfamiliar with the project, the about text and readme header, "Rust-Lightning, not Rusty's Lightning!", are somewhat opaque. It might be a good idea to change it to something that gets across the purpose of the project, like "A Low-level Lightning SDK in Rust".

### PR DESCRIPTION
**Pull request by casey**
- Change header to "Rust-Lightning"
- Add crate and documentation badges
- Add introductory paragraph
- Move previous intro content under "Status" heading

I wasn't sure where to put the old joke, but please feel free to push it to this PR branch.

Closes #728

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,17 @@
+Rust-Lightning
+==============
+
+[![Crate](https://img.shields.io/crates/v/lightning.svg?logo=rust)](https://crates.io/crates/lightning)
+[![Documentation](https://img.shields.io/static/v1?logo=read-the-docs&label=docs.rs&message=lightning&color=informational)](https://docs.rs/lightning/)
 [![Safety Dance](https://img.shields.io/badge/unsafe-forbidden-success.svg)](https://github.com/rust-secure-code/safety-dance/)
 
-Rust-Lightning, not Rusty's Lightning!
```
Comment (TheBlueMatt) [inline README.md:3]: We can keep things light, though, even if it's not the title :).

Comment (TheBlueMatt) [inline README.md:3]: Eh I'd say just move it down in the file and prefix with Tagline or something.

File: README.md [hunk 2]
```diff
@@ -1,9 +1,17 @@
...
-=====
+Rust-lightning is a low-level Bitcoin Lightning library written in Rust. It is
+low-level in that it does not handle networking, persistence, or any other I/O.
```
[the remark this answers is not in the document]
Comment (casey) [reply README.md]: I clarified this and added a reference to the `lightning-net-tokio` crate.

File: README.md (hunk 1 above)
Comment (casey) [reply README.md:3]: Added tagline section.

File: README.md [hunk 3]
```diff
@@ -1,9 +1,20 @@
...
 [![Safety Dance](https://img.shields.io/badge/unsafe-forbidden-success.svg)](https://github.com/rust-secure-code/safety-dance/)
 
-Rust-Lightning, not Rusty's Lightning!
-=====
+Rust-lightning is a Bitcoin Lightning library written in Rust. The main crate,
+`lightning`, does not handle networking, persistence, or any other I/O. Thus,
+it is runtime-agnostic, but users must implement basic networking logic, chain
+interactions, and disk storage.
+
+The `lightning-net-tokio` crate is a full Lightning using the
```
Comment (TheBlueMatt) [inline README.md:13]: We currently only implement the network-connection stuff. #681 does disk I/O (in a separate crate), and hopefully future ones will do chain sync.

**Review by TheBlueMatt — APPROVED** [review]
Looks good sans one note.

File: README.md (hunk 3 above)
Comment (casey) [reply README.md]: Fixed

File: README.md [hunk 4]
```diff
@@ -1,9 +1,20 @@
+Rust-Lightning
+==============
+
+[![Crate](https://img.shields.io/crates/v/lightning.svg?logo=rust)](https://crates.io/crates/lightning)
+[![Documentation](https://img.shields.io/static/v1?logo=read-the-docs&label=docs.rs&message=lightning&color=informational)](https://docs.rs/lightning/)
 [![Safety Dance](https://img.shields.io/badge/unsafe-forbidden-success.svg)](https://github.com/rust-secure-code/safety-dance/)
 
-Rust-Lightning, not Rusty's Lightning!
-=====
+Rust-lightning is a Bitcoin Lightning library written in Rust. The main crate,
```
Comment (naumenkogs) [inline README.md:8]: Maybe we could be pedantic and stick to either "Rust-lightning" or "Rust-Lightning"?
Cuz, you know "BitCoin" and all that :)

**Pull request merged**